src/common/utils/fhir/formatter.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { ProductDto } from 'src/common/dtos/product.dto';
+
+import * as formatterModule from './formatter';
+
+import { R4 } from '@ahryman40k/ts-fhir-types';
+
+const productDto: ProductDto = {
+  productCode: 'code',
+  productName: 'name',
+  mappings: [
+    { systemName: 'a', systemProductCode: 'code-a', productName: 'prod-a' },
+    { systemName: 'b', systemProductCode: 'code-b', productName: 'prod-b' },
+  ],
+  dateCreated: null,
+  lastUpdated: null,
+};
+
+describe('fhir formatter', () => {
+  describe('getProductsSystems', () => {
+    const getElement = jest.spyOn(formatterModule, 'getProductSystems');
+    it('should retrieve systems from product dto', () => {
+      const result = formatterModule.getProductSystems(productDto);
+      expect(result).toEqual(expect.arrayContaining(['a', 'b']));
+      expect(getElement).toHaveBeenCalled();
```

Comment:
By virtue of the prior test passing it implies _getProductSystems_ was called.

Reply:
yeah thanks for pointing that out, i made a mistake